File: backend/app/schemas/me.py
"""Response model for GET /me."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class MeResponse(BaseModel):
    model_config = ConfigDict(extra="ignore")

    device_id: str
    display_name: str | None
    points: int
    level: int
    level_name: str
    level_weight: int
    next_level_points: int   # -1 when at Legend (no further level)
    submission_count: int
    confirmation_count: int
    daily_streak: int
    can_rate_restaurants: bool
    first_seen: datetime

File: android/app/src/main/java/com/brokenlunch/gr/data/model/Me.kt
package com.brokenlunch.gr.data.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class MeResponse(
    @Json(name = "device_id") val deviceId: String,
    @Json(name = "display_name") val displayName: String?,
    @Json(name = "points") val points: Int,
    @Json(name = "level") val level: Int,
    @Json(name = "level_name") val levelName: String,
    @Json(name = "level_weight") val levelWeight: Int,
    @Json(name = "next_level_points") val nextLevelPoints: Int?,
    @Json(name = "submission_count") val submissionCount: Int,
    @Json(name = "confirmation_count") val confirmationCount: Int,
    @Json(name = "daily_streak") val dailyStreak: Int,
    @Json(name = "can_rate_restaurants") val canRateRestaurants: Boolean,
    @Json(name = "first_seen") val firstSeen: String,
)
